frontend/src/types.ts

```diff
@@ -318,17 +318,26 @@ export type PodContainer = {
   volumeMounts?: VolumeMount[];
 };
 
-export type PodStepState = {
-  stepName: string;
-  state: PodStepStateType;
-};
+export type PodStepState = { stepName: string; state: PodStepStateType };
 
 export enum PodStepStateType {
   success = 'Success',
   error = 'Error',
   loading = 'Loading',
 }
 
+export type PodContainerStatus = {
+  name?: string;
+  ready: boolean;
+  state?: {
+    running?: boolean | undefined;
+    waiting?: boolean | undefined;
+    terminated?: boolean | undefined;
+  };
+};
+
+export type PodContainerStatuses = (PodContainerStatus | undefined)[] | undefined;
```

Comment:
You want an array of container statuses or undefined? This seems like you failed to trim your array... why do you want this?